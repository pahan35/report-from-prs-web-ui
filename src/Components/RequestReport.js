import ReportModal from 'Components/ReportModal'
import {Button, Grid, GridCell, TextField} from 'mdc'
import React, {useState} from 'react'

export default function RequestReport() {
  const [forDays, setForDays] = useState(7)
  const [repo, setRepo] = useState()
  const [user, setUser] = useState()
  const [loading, setLoading] = useState(false)
  const [openModal, setOpenModal] = useState(false)

  const getReports = () => {
    setLoading(true)
    setTimeout(() => {
      setOpenModal(true)
      setLoading(false)
    }, 3500)
  }

  return (
    <Grid>
      <GridCell align="middle" span={12}>
        <TextField
          className="pr-field"
          label="User"
          value={user}
          onChange={e => setUser(e.target.value)}
        />
        <TextField
          className="pr-field"
          label="Repo"
          value={repo}
          onChange={e => setRepo(e.target.value)}
        />
        <TextField
          className="pr-field"
          label="For days"
          type="number"
          value={forDays}
          onChange={e => setForDays(e.target.value)}
        />
      </GridCell>
      <GridCell align="middle" span={4}>
        <Button
          disabled={loading}
          onClick={() => {
            getReports()
          }}
        >
          Get Reports
        </Button>
      </GridCell>
      <ReportModal open={openModal} onClose={() => setOpenModal(false)} />
    </Grid>
  )
}
