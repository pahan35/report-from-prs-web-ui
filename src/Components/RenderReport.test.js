import {render} from '@testing-library/react'
import RequestReport from 'Components/RequestReport'
import React from 'react'
import renderer from 'react-test-renderer'

test('RequestReport snapshot', () => {
  const tree = renderer.create(<RequestReport />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('RequestReport', () => {
  const {getByLabelText} = render(<RequestReport />)
  expect(getByLabelText('Repo')).toBeInTheDocument()
})
