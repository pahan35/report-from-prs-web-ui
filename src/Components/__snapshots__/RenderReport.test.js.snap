// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestReport snapshot 1`] = `
<div
  className="mdc-layout-grid"
>
  <div
    className="mdc-layout-grid__inner"
  >
    <div
      className="mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-12"
    >
      <div
        className="pr-field pr-field mdc-ripple-surface mdc-text-field mdc-text-field--upgraded"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <input
          className="mdc-text-field__input"
          id="textfield-test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          type="text"
        />
        <label
          className="mdc-floating-label"
          htmlFor="textfield-test"
          onAnimationEnd={[Function]}
          style={Object {}}
        >
          User
        </label>
        <div
          className="mdc-line-ripple "
          onTransitionEnd={[Function]}
          style={Object {}}
        />
      </div>
      <div
        className="pr-field pr-field mdc-ripple-surface mdc-text-field mdc-text-field--upgraded"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <input
          className="mdc-text-field__input"
          id="textfield-test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          type="text"
        />
        <label
          className="mdc-floating-label"
          htmlFor="textfield-test"
          onAnimationEnd={[Function]}
          style={Object {}}
        >
          Repo
        </label>
        <div
          className="mdc-line-ripple "
          onTransitionEnd={[Function]}
          style={Object {}}
        />
      </div>
      <div
        className="pr-field pr-field mdc-ripple-surface mdc-text-field mdc-text-field--upgraded"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <input
          className="mdc-text-field__input"
          id="textfield-test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          type="number"
          value={7}
        />
        <label
          className="mdc-floating-label"
          htmlFor="textfield-test"
          onAnimationEnd={[Function]}
          style={Object {}}
        >
          For days
        </label>
        <div
          className="mdc-line-ripple "
          onTransitionEnd={[Function]}
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-4"
    >
      <button
        className="mdc-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <span
          className="mdc-button__label"
        >
          Get Reports
        </span>
      </button>
    </div>
    <div
      aria-modal={true}
      className="mdc-dialog"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="mdc-dialog__container"
      >
        <div
          className="mdc-dialog__surface"
        >
          <h2
            className="mdc-dialog__title"
          >
            Reports
          </h2>
          <div
            className="mdc-dialog__content"
          >
            No reports yet
          </div>
        </div>
      </div>
      <div
        className="mdc-dialog__scrim"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
